modules/sandboxed-containers-check-node-eligiblilty.adoc

```diff
@@ -0,0 +1,116 @@
+// Module included in the following assemblies:
+//
+// * sandboxed_containers/deploying-sandboxed-container-workloads.adoc
+
+:_module-type: PROCEDURE
+[id="sandboxed-containers-check-node-eligiblilty_{context}"]
+= Checking whether cluster nodes are eligible to run {sandboxed-containers-first}
+
+Before running {sandboxed-containers-first}, you can check whether the nodes in your cluster are eligible to run Kata containers. Some cluster nodes might not comply with sandboxed containers' minimum requirements. The most common reason for node ineligibility is the lack of virtualization support on the node. If you attempt to run sandboxed workloads on ineligible nodes, you will experience errors. You can use the Node Feature Discovery Operator (NFD) and a `NodeFeatureDiscovery` resource to automatically check node eligibility.
+
+[NOTE]
+====
+If you want to install the Kata runtime on only selected worker nodes that you know are eligible, apply the `feature.node.kubernetes.io/runtime.kata=true` label to the selected nodes and set `checkNodeEligibility: true` in the `KataConfig` resource.
+
+Alternatively, to install the Kata runtime on all worker nodes, set `checkNodeEligibility: false` in the `KataConfig` resource.
```

Comment:
Thanks, this looks good already!  I'd just consider adding a sentence or two here to make sure users understand that this is not an administrative decision that's purely up to them to decide if they want to run kata on a half or a quarter of their workers (KataConfig.spec.kataConfigPoolSelector serves that purpose) and they should only do this if they are sure by external means that all workers are eligible to run kata.

This is to prevent confusion in cases like clusters where e.g. workers are virtual machines on AWS which cannot run kata due to lack of virtualisation support. In such cases, installation will still succeed as it's basically just installation of a bunch of .rpms that won't be disturbed by eligibility issues. Errors will only come up when actually attempting to run sandboxed payloads on such nodes, and while I admittedly don't remember the exact error messages (@bpradipt ?) I seem to remember they aren't super obvious as to what the root cause is.

Reply:
Cool thanks. Will update. Perhaps some detail as to the error messages would be useful? @bpradipt can you help with that?